src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,268 @@
+```bash
+python3 app.py
+```
+Your REST API is now up and running on `http://127.0.0.1:5000`
+
+## Testing your API
+You can test your API using tools like Postman or cURL.
+
+### List all tasks
+You can use the following curl command to list all the avaliable tasks:
+```bash
+curl http://127.0.0.1:5000/tasks
+```
+Expected Result:
+```json
+{
+  "tasks": [
+    {
+      "completed": false,
+      "id": 1,
+      "title": "Buy groceries"
+    },
+    {
+      "completed": true,
+      "id": 2,
+      "title": "Clean the house"
+    }
+  ]
+}
+```
+
+### List a specific task
+To retrieve a specific task by its ID, you can use the following curl command, replacing `<task_id>` with the id of the task you want to retrieve:
+```bash
+curl http://127.0.0.1:5000/tasks/<task_id>
+```
+Example:
+```bash
+curl http://127.0.0.1:5000/tasks/1
+```
+Expected Result:
+```json
+{
+  "task": {
+    "completed": false,
+    "id": 1,
+    "title": "Buy groceries"
+  }
+}
+```
+
+### Create a New Task
+To create a new task, you can use following curl command, replacing `<task_data>` with the JSON data for the new task:
+```bash
+curl -X POST -H "Content-Type: application/json" -d '<json_data>' http://127.0.0.1:5000/tasks
+```
+Example:
+```bash
+curl -X POST -H "Content-Type: application/json" -d '{"id":3,"task":"Finish the report","completed":false}' http://127.0.0.1:5000/tasks
+```
+Expected Result:
+```json
+{
+  "task": {
+    "completed": false,
+    "id": 3,
+    "task": "Finish the report"
+  }
+}
+```
+
+### Update a task
+To update an existing task, use the following curl command, replacing `<task_id>` with the id of the task, and `<updated_data>` with the JSON data you want to update:
+```bash
+curl -X PUT -H "Content-Type: application/json" -d '<updated_data>' http://127.0.0.1:5000/tasks/<task_id>
+```
+Example:
+```bash
+curl -X PUT -H "Content-Type: application/json" -d '{"completed":true}' http://127.0.0.1:5000/tasks/3
+```
+Expected Result:
+```json
+{
+  "task": {
+    "completed": true,
+    "id": 3,
+    "task": "Finish the report"
+  }
+}
+```
+
+### Delete a task
+To delete a task, use the following curl command, replacing `<task_id>` with the id of the task you want to delete:
+```bash
+curl -X DELETE http://127.0.0.1:5000/tasks/<task_id>
+```
+Example:
+```bash
+curl -X DELETE http://127.0.0.1:5000/tasks/3
+```
+Expected Result:
+```json
+{
+  "result": "Task deleted"
+}
+```
+
+Let's try to retrieve a task which doesn't exist in the list of task. For instance, we just deleted task with id = 3. Now, let's try to retrieve the task with id = 3 and see the result.
+```bash
+curl http://127.0.0.1:5000/tasks/3
+```
+Expected result:
+```json
+{
+  "error": "Task not found"
+}
+```
+We see that the JSON response returns the error `Task not found`, which indicates that the task doesn't exist in the list of tasks.
+
+Now, let's look into the cases with invalid requests:
+1. **Invalid endpoint**:
```

Comment:
I think so far you haven't really written anything about endpoints at all... Since your blog seems to be aimed at beginners, it's would be better to introduce what's an endpoint earlier in your text